### ISSUE TITLE
Compile time benchmarks in CI?

### ISSUE THREAD

**Issue by Kmeakin** [yeslogic/fathom#432]
I've submitted some PRs in the past that remove some inefficiencies in the compiler (https://github.com/yeslogic/fathom/pull/427, https://github.com/yeslogic/fathom/pull/426). In those cases, they were obvious performance wins, but in the future we might want to include compile time benchmarks in CI, so we can track the long term tend in compile times and catch regressions/test performance improvements.

Information on bench marking in CI is scarce. Criterion [recommends](https://bheisler.github.io/criterion.rs/book/iai/iai.html) [iai](https://github.com/bheisler/iai). We should also investigate what other compilers (eg rustc, cranelift) use.

**Comment by brendanzab** [comment]
Yeah, this would be nice! I need to be better at profiling as well to be honest. Stuff like memory usage and performance of binary parsing would be nice as well perhaps?

I think I’ve also seen some places that it’s worth taking CI-based profiling with a grain of salt due to it being run on shared infrastructure. Curious to know about best practices but it still might be a useful thing to do at any rate.
